worker/durable-objects/PresenceRoom.ts

```
@@ -0,0 +1,122 @@
+    let parsed: ClientFrame | null = null;
+    try { parsed = JSON.parse(raw) as ClientFrame; } catch { return; }
+    if (!parsed || typeof parsed !== 'object') return;
+    if (parsed.type === 'ping') {
+      // Echo a snapshot so latency-sensitive clients can measure rtt + refresh.
+      this.sendFrameTo(ws, this.buildSnapshotFrame());
+    }
+    // 'identify' is a no-op on the wire today — userId is locked at the URL
+    // query string and stored as a tag during accept. Reserved for future use.
+  }
+
+  webSocketClose(ws: WebSocket, _code: number, _reason: string, _wasClean: boolean): void {
```

ESLint

[error] 'WebSocket' is not defined.

(no-undef)